src/Draco.Repl/ReplPromptCallbacks.cs

```diff
@@ -97,6 +83,20 @@ protected override Task<IReadOnlyList<CompletionItem>> GetCompletionItemsAsync(
         return Task.FromResult<IReadOnlyList<CompletionItem>>(result);
     }
 
+    protected override Task<bool> ShouldOpenCompletionWindowAsync(
+        string text, int caret, KeyPress keyPress, CancellationToken cancellationToken)
+    {
+        var script = this.MakeScript(text);
+        var tree = script.Compilation.SyntaxTrees.Single();
+        var semanticModel = script.Compilation.GetSemanticModel(tree);
+
+        var cursorPosition = tree.IndexToSyntaxPosition(caret);
+
+        var completionItems = this.completionService.GetCompletions(semanticModel, caret);
+
+        return Task.FromResult(completionItems.Any());
+    }
```

Comment:
Unrelated to the PR but, that's one thing I find sad with .NET async.  
async keyword not being able to come up with that is sad and make the user write more convoluted code.

Reply:
Oh yes I hate this. But explicit `async` is a warning...